components/tidb_query_datatype/src/codec/datum.rs

```diff
@@ -374,6 +375,7 @@ impl Datum {
             Datum::Time(t) => Some(!t.is_zero()),
             Datum::Dur(d) => Some(!d.is_zero()),
             Datum::Dec(d) => Some(ConvertTo::<f64>::convert(&d, ctx)?.round() != 0f64),
+            Datum::Json(j) => Some(j.as_ref().as_mysql_bool(ctx).unwrap()),
```

Comment:
Let's replace `unwrap()` with `?` here, in case that future work makes `as_mysql_bool` returns Err.

Reply:
Addressed.